### ISSUE TITLE
VPC Peering Connection Failing Because Cross-Region Error

### ISSUE THREAD

**Issue by brianpham** [terraform-providers/terraform-provider-aws#8623]
### Terraform Version

Terraform v0.11.13
+ provider.aws v2.10.0

### Affected Resource(s)

* aws_XXXXX

### Terraform Configuration Files

```hcl
# Requester's side of the connection.
resource "aws_vpc_peering_connection" "us-east_us-west-peer-request" {
  vpc_id        = "${data.aws_vpc.vpc-us-east-1.id}"
  peer_vpc_id   = "${data.aws_vpc.vpc-us-west-1.id}"
  peer_owner_id = "${data.aws_caller_identity.identity-us-west-1.account_id}"
  peer_region   = "us-west-1"
  auto_accept   = false

  tags = {
    Side = "Requester"
  }
}

# Accepter's side of the connection.
resource "aws_vpc_peering_connection_accepter" "us-east_us-west-peer-accept" {
  provider                  = "aws.us-west-1"
  vpc_peering_connection_id = "${aws_vpc_peering_connection.us-east_us-west-peer-request.id}"
  auto_accept               = true

  tags = {
    Side = "Accepter"
  }
}

resource "aws_vpc_peering_connection_options" "us-east_us-west-peer-options" {
  vpc_peering_connection_id = "${aws_vpc_peering_connection_accepter.us-east_us-west-peer-accept.id}"

  accepter {
    allow_remote_vpc_dns_resolution = true
  }

  requester {
    allow_remote_vpc_dns_resolution = true
  }
}
```

### Debug Output

https://gist.github.com/brianpham/6d8f1c7a519abaccf32612dc4c73a805

### Expected Behavior

VPC Peering should be created with the dns options set to true for the accepter and requester. 

### Actual Behavior

VPC Peering is failing because of the following error 
```Error modifying VPC Peering Connection Options: OperationNotPermitted: Modifying VPC peering connection options AllowEgressFromLocalClassicLinkToRemoteVpc, AllowEgressFromLocalVpcToRemoteClassicLink is not supported for cross-region VPC peering connections```

### Steps to Reproduce

1. `terraform apply` using a regular VPC peering setup

### References

https://github.com/terraform-providers/terraform-provider-aws/pull/3909

* #0000

**Comment by brianpham** [comment]
Closing. Looks like this is fixed in https://github.com/terraform-providers/terraform-provider-aws/pull/7627

**Comment by tprakash17** [comment]
#7627 still open btw ...changes not merged yet.

**Comment by krish7919** [comment]
This is still an ongoing issue...

**Comment by ashish30jain** [comment]
With latest provider 2.30 as well this is an issue.

**Comment by bflad** [comment]
#7627 was recently merged and will release with version 2.31.0 of the Terraform AWS Provider on Thursday.
